config/resolve_credentials.go

```diff
@@ -384,25 +384,36 @@ func assumeWebIdentity(ctx context.Context, cfg *aws.Config, filepath string, ro
 		return fmt.Errorf("token file path is not set")
 	}
 
-	if len(roleARN) == 0 {
-		return fmt.Errorf("role ARN is not set")
-	}
-
-	optFns := []func(*stscreds.WebIdentityRoleOptions){
-		func(options *stscreds.WebIdentityRoleOptions) {
-			options.RoleSessionName = sessionName
-		},
-	}
+	var optFns []func(*stscreds.WebIdentityRoleOptions)
 
 	optFn, found, err := getWebIdentityCredentialProviderOptions(ctx, configs)
 	if err != nil {
 		return err
 	}
+
 	if found {
 		optFns = append(optFns, optFn)
 	}
 
-	provider := stscreds.NewWebIdentityRoleProvider(sts.NewFromConfig(*cfg), roleARN, stscreds.IdentityTokenFile(filepath), optFns...)
+	opts := stscreds.WebIdentityRoleOptions{
+		RoleARN:         roleARN,
+		RoleSessionName: sessionName,
+	}
```

Comment:
The `opts` value isn't passed to `stscreds.NewWebIdentityRoleProvider`, so the `RoleSessionName` set here is lost.

The original declaration of `optFns` should be restored, i.e.

```go
optFns := []func(*stscreds.WebIdentityRoleOptions){
	func(options *stscreds.WebIdentityRoleOptions) {
		options.RoleSessionName = sessionName
	},
}
```

Reply:
Good catch, I've restored the way `RoleSessionName` was being set.